packages/slice-machine/lib/builders/common/EditModal/index.js

```diff
@@ -165,10 +166,7 @@ const EditModal = ({ close, data, fields, onSave, getFieldMockConfig }) => {
           })();
 
           // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-          const updatedValue = {
-            ...initialModelValues,
-            ...value,
-          };
+          const updatedValue = merge(initialModelValues, value);
```

Comment:
interesting bug: 
initialModelValues was `{type: "...", config: {...}}` (coming from the model state from the BE)
value has the same type but its config contains the data from the edit modal (therefore would not have the property `labels`).

If `lables` was put manually, since we were doing a shallow merge copy, we would override the old `config` (with labels) with the new one (without labels)